### DISCUSSION TITLE
seaweedfs的8888服务安全

### DISCUSSION THREAD

**Discussion by userStarlight** [seaweedfs/seaweedfs#4464] [General]
seaweedfs的8888端口号的WebUI可以禁用吗？如果可以该如何禁用呢，希望只禁用访问http://localhost:8888的web页面，但是还是可以通过http://localhost:8888/file.jpg来访问文件，这个可以实现吗？

**Comment by chrislusf** [comment]
`weed filer -disableDirListing`

**Reply by userStarlight** [reply]
感谢您使用宝贵的时间回复
我是使用一下方式启动的Seaweedfs
version: '3.1'
services:
  fs:
    restart: always
    image: chrislusf/seaweedfs
    container_name: fs
    command: server -s3
    ports:
      - 8888:8888
      - 9333:9333
 
有点不明白是这样输入命令吗？
/ # weed filer -disableDirListing
W0513 02:46:36.110088 filer_server.go:143 skipping default store dir in ./filerldb2
I0513 02:46:36.111086 leveldb2_store.go:42 filer store leveldb2 dir: /data/filerldb2
I0513 02:46:36.111138 file_util.go:23 Folder /data/filerldb2 Permission: -rwxr-xr-x
E0513 02:46:36.111208 leveldb2_store.go:62 filer store open dir /data/filerldb2/00: resource temporarily unavailable
F0513 02:46:36.111413 configuration.go:25 failed to initialize store for leveldb2: resource temporarily unavailable
goroutine 1 [running]:
github.com/seaweedfs/seaweedfs/weed/glog.stacks(0x0)
        /go/src/github.com/seaweedfs/seaweedfs/weed/glog/glog.go:768 +0x89
github.com/seaweedfs/seaweedfs/weed/glog.(*loggingT).output(0x5024400, 0x3, 0xc0007f0380, {0x3f661e9?, 0x10?}, 0x2?, 0x0)
        /go/src/github.com/seaweedfs/seaweedfs/weed/glog/glog.go:719 +0x3aa
github.com/seaweedfs/seaweedfs/weed/glog.(*loggingT).printf(0x2ff9bbf?, 0x8?, {0x3076513, 0x26}, {0xc0004f9818, 0x2, 0x2})
        /go/src/github.com/seaweedfs/seaweedfs/weed/glog/glog.go:657 +0x10f
github.com/seaweedfs/seaweedfs/weed/glog.Fatalf(...)
        /go/src/github.com/seaweedfs/seaweedfs/weed/glog/glog.go:1154
github.com/seaweedfs/seaweedfs/weed/filer.(*Filer).LoadConfiguration(0xc0007f9220, 0x501a400)
        /go/src/github.com/seaweedfs/seaweedfs/weed/filer/configuration.go:25 +0x2fb
github.com/seaweedfs/seaweedfs/weed/server.NewFilerServer(0xc000455e40, 0xc000455e40, 0xc0006ae460)
        /go/src/github.com/seaweedfs/seaweedfs/weed/server/filer_server.go:153 +0x97f
github.com/seaweedfs/seaweedfs/weed/command.(*FilerOptions).startFiler(0x50248e0)
        /go/src/github.com/seaweedfs/seaweedfs/weed/command/filer.go:235 +0x386
github.com/seaweedfs/seaweedfs/weed/command.runFiler(0x4ffe798?, {0xc0000501a0?, 0x1?, 0x1?})
        /go/src/github.com/seaweedfs/seaweedfs/weed/command/filer.go:211 +0x4df
main.main()
        /go/src/github.com/seaweedfs/seaweedfs/weed/weed.go:82 +0x3a3
/ #
如果是以上方式输入命令，我在重启docker后还是可以通过 http://localhost:8888 访问到webUI页面，请问这条命令该如何使用

**Reply by userStarlight** [reply]
感谢您使用宝贵的时间回复，目前已经解决了问题，是我使用命令的方式不对
version: '3.1'
services:
  fs:
    restart: always
    image: chrislusf/seaweedfs
    container_name: fs
    command: server -s3 -filer.disableDirListing=true
    ports:
      - 8888:8888
改为这种方式启动就可以禁止用户通过http://locatlhost:8888来访问WebUI了
